src/os/newstore/NewStore.cc

```diff
@@ -3299,8 +3327,8 @@ int NewStore::_do_write_all_overlays(TransContext *txc,
         bl_next_data.substr_of(bl_next, next->second.value_offset,
```

Comment:
No, this isn't trading read for write. We have to read all of the overlays anyway, either doing combination or not. Actually this is the change in PR 4464, not this one.